src/modules/dom.js

```diff
@@ -0,0 +1,71 @@
+export const searchForm = document.querySelector('.control');
+export const checkCity = document.getElementById('cityInput');
+export const errorMessage = document.getElementById('warn');
+
+const weatherImg = document.querySelector('.icon');
+const desc = document.querySelector('.weather-status');
+const currTemp = document.querySelector('.main-temp');
+const currCity = document.querySelectorAll('.location>span');
+const moreAbt = document.querySelectorAll('.more-info > p');
+const displayCity = document.getElementById('weather-forecast');
+const currDate = document.querySelector('.date');
+const c = document.getElementById('switchF');
+const d = document.getElementById('degrees');
+
+export default (data, tempF = null) => {
+  const list = document.createElement('li');
+  list.classList.add('city');
+  currCity[0].innerHTML = data.name;
+  currCity[1].innerHTML = data.country;
+  currDate.innerHTML = new Date();
+  currTemp.innerHTML = `<i class='fa fa-thermometer-three-quarters'></i>${Math.round(
+    data.temp,
+  )}°C`;
+  weatherImg.src = `https://openweathermap.org/img/wn/${data.icon}@2x.png`;
+  desc.innerHTML = data.description;
+  let currTempF = false;
+  const toggleTempF = (el, cc, tempInfo) => {
+    const cel = `<i class='fa fa-thermometer-three-quarters'></i>${Math.round(
+      tempInfo,
+    )}°C`;
+    const elTempF = `<i class='fa fa-thermometer-three-quarters'></i>${Math.round(
+      tempF,
+    )}°F`;
+    if (currTempF) {
+      cc.innerHTML = cel;
+      el.innerHTML = 'Switch to °F';
+      currTempF = false;
+    } else {
+      cc.innerHTML = elTempF;
+      el.innerHTML = 'Switch to °C';
+      currTempF = true;
+    }
+  };
+  moreAbt[0].innerHTML = `
+  <span class='b-feel'>Feels Like</span>
+  ${Math.round(data.feelsLike)}°C
+  `;
+  moreAbt[1].innerHTML = `
+  <i class='fa fa-temperature-high'></i>
+  <span class='b-feel'>Max temp</span>${Math.round(
+    data.tempMax,
+  )}°C, <i class='fa fa-temperature-low'></i>
+  <span class='b-feel'>Min temp</span>${Math.round(data.tempMin)}°C
+  `;
+  d.style.display = 'block';
+  d.addEventListener('click', () => {
+    toggleTempF(d, c, data.temp);
+  });
+  displayCity.className = 'show';
+};
+
+export const selectImage = (cityName) => {
+  const script = document.createElement('script');
+  script.src = `https://api.flickr.com/services/feeds/photos_public.gne?format=json&jsoncallback=abc&tags=${cityName}`;
+  document.head.appendChild(script);
+  window.abc = function abc(data) {
+    const body = document.querySelector('body');
+    body.style.background = `url('${data.items[3].media.m}') no-repeat`;
+    body.style.backgroundSize = 'cover';
```

Comment:
Why not have this on your stylesheet? No need to reset it on every call.

Reply:
Hello @MauricioRobayo, thank you for the feedback, this is my first time to work with promises and I actually did not know anything about Promise.all... I am however confused now because the Fahrenheit when I click on the button to toggle the temperatures, returns 0°F. The metric is working fine but not the imperial. Could you please enlighten me more about this? Thank you.
~~~
const fetchWeather = async (city, unit) => {
    const response = await fetch(
      `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=${unit}`
    );
    return response.json();
  };

  Promise.all([
    fetchWeather(currentCity, "metric"),
    fetchWeather(currentCity, "imperial")
  ]).then(([celsiusData, FahrenheitData]) => {
    updateDOM(parseJSON(celsiusData), parseJSON(FahrenheitData).tempF);
  });
~~~